lua/nvim-lsp-installer/servers/gopls.lua

```diff
@@ -0,0 +1,29 @@
+local server = require('nvim-lsp-installer.server')
+
+local root_dir = server.get_server_root_path('go')
+
+local install_cmd = [=[
+GO111MODULE=on GOBIN="$PWD" GOPATH="$PWD"  go get golang.org/x/tools/gopls@latest;
+
+if ! command -v ./gopls &> /dev/null; 
+then 
+  echo "something went wrong!"
+else 
+  echo "installation completed"
+fi
+
+]=]
+
+return server.Server:new {
+  name = "gopls",
+  root_dir = root_dir,
+  pre_install_check = function ()
+    if vim.fn.executable("go") ~= 1 then
+      error("Please install the Go CLI before installing gopls (https://golang.org/doc/install).")
+    end
+  end,
+  install_cmd = install_cmd,
+  default_options = {
```

Comment:
Also, just a reference for posterity, the server doesn't start unless a `.git` or a `go.mod` is found (as shown [here](https://github.com/neovim/nvim-lspconfig/blob/master/CONFIG.md#gopls)), which prevents the server to be used, for example, in a main.go in your $HOME (unless itself is a go.mod or contains .git) and I think this behaviour can be changed adding 
`root_dir = root_pattern("go.mod", ".git") or dirname`
to the config. I write this here just because I was clueless at first.